updates after ARC review (#144)
riscv/riscv-security-model: specification/src/chapter3.adoc

| SR_HYP_001
| For virtualised environments, Hypervisor extension MUST be supported

| SR_HYP_002
| For non-virtualised environments, Hypervisor extension MUST be supported
|===

Alternatively SPMP can be used instead of address translation and protection to support static partitioning
hypervisors, for example on systems with hard and deterministic real time requirements.

> MMU, PMP/Smepmp, and SPMP are discussed later in this chapter.

Reviewer: ```suggestion
Address translation and protection, PMP/Smepmp, and SPMP are discussed later in this chapter.
```

| SR_PMP_006
| SPMP configurations MUST only be directly accessible to M-Mode and S-Mode

| SR_PMP_007
| vSPMP configurations MUST only be directly accessible to M-Mode and HS-Mode and VS-mode. 

| SR_PMP_008
| vSPMP MUST be used to protect VS-mode from VU-Mode.

| SR_PMP_009
| vSPMP MUST be used to protect VU-mode workloads from other VU-Mode workloads.

|===

> NOTE: Dependent on the use case, it may be recommended or mandatory to use SPMP to protect S-Mode from lower privilege levels (with or without the H-extension). Address translation and proteciton may be an additional or alternative mechanism.

Reviewer: ```suggestion
NOTE: Dependent on the use case, it may be recommended or mandatory to use SPMP to protect S-Mode from lower privilege levels (with or without the H-extension). Address translation and protection may be an additional or alternative mechanism.
```

NOTE: Dependent on the use case, Trusted Execution Environments may require or recommend the use of SPMP to isolate trusted applications from each other.
Address translation and protection may be an additional or alternative mechanism

[[atp]]

> == Address translation and protection

Author: ```suggestion
=== Address translation and protection
```

> === Address translation and protection

Author: ```suggestion
==== Address translation and protection
```

> *See sections 4.3 to 4.6 (Page-Based Virtual-Memory Systems) of the https://github.com/riscv/riscv-isa-manual/releases/latest[Privileged

Author: ```suggestion
*See section Page-Based Virtual-Memory Systems of the https://github.com/riscv/riscv-isa-manual/releases/latest[Privileged
```

ISA] specification.*

_Address translation and protection_ enables:

* Isolating an OS from workloads, and workloads from each other on a system without H-extension (one-stage
translation)
* Isolating a hypervisor from a guest, on a system with H-extension (two-stage
translation)

[#cat_sr_sub_atp]
[width=100%]
[%header, cols="5,20"]
|===
| ID#
| Requirement

| SR_ATP_001
| If the Sv extension is supported then 1st-stage page tables MUST be used to protect
the S-Mode Supervisor domain from accesses made by U-Mode.

| SR_ATP_002
| If the H extension is supported 1st-stage and/or G-stage page tables MUST be used to protect
Supervisor domain H/S-Mode from lower privilege levels.

| SR_ATP_003
| Address translation and protection MUST be used to protect resources assigned to one workload from other workloads
|===

> NOTE: Dependent on the use case, it may be recommended or mandatory to use address trnslation and protection to protect S-Mode from lower privilege levels (with or without the H-extension). The SPMP and PMP may be an additional or alternative mechanism.

Reviewer: ```suggestion
NOTE: Dependent on the use case, it may be recommended or mandatory to use address translation and protection to protect S-Mode from lower privilege levels (with or without the H-extension). The SPMP and PMP may be an additional or alternative mechanism.
```

> NOTE: Dependent on the use case, Trusted Execution Environments may require or recommend the use of address trnslation and protection to isolate trusted applications from each other. The SPMP and PMP may be an additional or alternative mechanism.

Reviewer: ```suggestion
NOTE: Dependent on the use case, Trusted Execution Environments may require or recommend the use of address translation and protection to isolate trusted applications from each other. The SPMP and PMP may be an additional or alternative mechanism.
```
